src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/LicenseAcceptanceWindow.xaml.cs

```diff
@@ -72,5 +75,24 @@ private void OnButtonKeyDown(object sender, KeyEventArgs e)
                 DialogResult = false;
             }
         }
+
+        private void LicenseAcceptanceWindow_Loaded(object sender, RoutedEventArgs e)
+        {
+            var screen = Screen.FromPoint(new System.Drawing.Point((int)Left, (int)Top));
+            var dpiScale = VisualTreeHelper.GetDpi(this);
+
+            double screenWidth = screen.WorkingArea.Width / dpiScale.DpiScaleX;
+            double screenHeight = screen.WorkingArea.Height / dpiScale.DpiScaleY;
+
+            int desiredLength = 450;
+            int heightPadding = 25; // Make window height at least this much smaller than the screen height.
+            Height = Math.Min(desiredLength, screenHeight - heightPadding);
+            MaxHeight = screenHeight;
+            MinHeight = Height;
+
+            Width = Math.Min(desiredLength, screenWidth);
+            MaxWidth = screenWidth;
+            MinWidth = Width;
```

Comment:
Should we limit the max size?  Do we care about a scenario where the customer moves the window from one screen to another, and the screen resolution might be different? If we let customers resize to any size they want, then it's their responsibility to "fix" the size if they accidentally make it too large or small and can no longer read the contents.

Similarly for min sizes, it would be strange for someone to resize it to only 10 pixels high so nothing is readable, not even the title bar. But I'm more understanding. Still, I feel like if we want a min height/width, can we find a suitable size for the smallest of resolutions at the highest text sizes, and just use that always? The default size when the window is opened doesn't have to equal the min size.

Reply:
Min size already matches default size today https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/LicenseAcceptanceWindow.xaml#L11. This is also correct per DevDiv guidelines: "Resizable dialogs must have a minimum size".

Regarding Max sizes, I can't think of a reason why we should support making the window larger than the screen. There's no guideline for this, so I think worrying about it is unnecessary. My experience is most dialogs don't work well moved around, since scaling and other factors weren't applied at render-time.